javascript-carousel/main.js

```diff
@@ -0,0 +1,117 @@
+const img = document.querySelector('img');
+let currentIndex = 0;
+let intervalID = 0;
+
+const arrObj = [
+  {
+    imgSrc: 'images/001.png',
+    dotID: 'c1',
+  },
+  {
+    imgSrc: 'images/004.png',
+    dotID: 'c2',
+  },
+  {
+    imgSrc: 'images/007.png',
+    dotID: 'c3',
+  },
+  {
+    imgSrc: 'images/025.png',
+    dotID: 'c4',
+  },
+  {
+    imgSrc: 'images/039.png',
+    dotID: 'c5',
+  },
+];
+
+// *************** auto play ***********************************
+function autoPlay(getUpdatedValue) {
+  intervalID = setInterval(() => {
+    // update the currentIndex
+    // NOTE: the initial image and filled circle has been set already, so skip to the next index
+    currentIndex = currentIndex + 1 <= arrObj.length - 1 ? currentIndex + 1 : 0;
+    getUpdatedValue(currentIndex);
+    // show the image and filled circle
+    showImgAndFilledCircle();
+    // change filled circle to open circle
+    showOpenCircles();
+  }, 3000);
+}
+
+// ************* Get updated async current index  **********************
+function getUpdatedValue(updatedValue) {
+  return updatedValue;
+}
+autoPlay(getUpdatedValue);
+
+// ************ Clicking the previous or the next arrow button *****************
+const nextButton = document.querySelector('.forward');
+const previousButton = document.querySelector('.backward');
+
+nextButton.addEventListener('click', nextImg);
+previousButton.addEventListener('click', previousImg);
+
+function nextImg() {
+  clearInterval(intervalID);
+  currentIndex = currentIndex + 1 <= arrObj.length - 1 ? currentIndex + 1 : 0;
+  showImgAndFilledCircle();
+  showOpenCircles();
+  autoPlay(getUpdatedValue);
+}
```

Comment:
`autoPlay` would be more self-contained if it handled clearing the previous interval internally.